src/Security/RememberLoginHash.php

```diff
@@ -190,14 +204,21 @@ public static function generate(Member $member)
     /**
      * Generates a new hash for this member but keeps the device ID intact
      *
+     * @deprecated 5.3.0 Token renewal will be removed in 6.0.0
      * @return RememberLoginHash
      */
     public function renew()
     {
-        $hash = $this->getNewHash($this->Member());
-        $this->Hash = $hash;
+        // Only regenerate token if configured to do so
+        $replaceToken = RememberLoginHash::config()->get('replace_token_during_session_renewal');
+        if ($replaceToken) {
+            $hash       = $this->getNewHash($this->Member());
+            $this->Hash = $hash;
+        }
+
         $this->extend('onAfterRenewToken');
```

Comment:
```suggestion
        $this->extend('onAfterRenewToken', $replaceToken);
```
This gives people a way to tell if their token was actually renewed, so they can opt to not do their extension logic if it should only be done when the token is actually renewed.

This doesn't break BC because there's no requirement that the extension method accepts any arguments.